ci/generate-and-upload-build-steps.sh

```diff
@@ -2,11 +2,15 @@
 set -euo pipefail
 
 upload_build_configuration_step() {
-    export ENGINE_COMMIT_HASH=$1
-    export BUILD_PLATFORM=$2
-    export BUILD_TARGET=$3
-    export BUILD_STATE=$4
-    buildkite-agent pipeline upload "ci/gdk_build.template.steps.yaml"
+    export ENGINE_COMMIT_HASH=${1}
+    export BUILD_PLATFORM=${2}
+    export BUILD_TARGET=${3}
+    export BUILD_STATE=${4}
+    if [[ ${BUILD_PLATFORM} == "Mac" ]]; then
+        buildkite-agent pipeline upload "ci/gdk_build_macos.template.steps.yaml"
```

Comment:
I see @jamiebrynes7 has been virally spreading his dynamic pipeline mutations... ;)

Reply:
haha, I don't think this is jamie's fault ;)